lib/ts/recipe/totp/api/implementation.ts

```diff
@@ -0,0 +1,153 @@
+import { getUser } from "../../..";
+
+export default function getAPIInterface(): APIInterface {
+    return {
+        createDevicePOST: async function ({ deviceName, options, session, userContext }) {
+            const userId = session.getUserId();
+
+            let userIdentifierInfo: string | undefined = undefined;
+            const emailOrPhoneInfo = await TotpRecipe.getInstanceOrThrowError().getUserIdentifierInfoForUserId(
+                session.getUserId(),
+                userContext
+            );
+            if (emailOrPhoneInfo.status === "OK") {
+                userIdentifierInfo = emailOrPhoneInfo.info;
+            } else if (emailOrPhoneInfo.status === "UNKNOWN_USER_ID_ERROR") {
+                throw new SessionError({
+                    type: SessionError.UNAUTHORISED,
+                    message: "Unknown User ID provided",
+                });
+            } else if (emailOrPhoneInfo.status === "USER_IDENTIFIER_INFO_DOES_NOT_EXIST_ERROR") {
+                // Ignore since UserIdentifierInfo is optional
+            }
+
+            let mfaInstance = MultiFactorAuthRecipe.getInstance();
+
+            if (mfaInstance === undefined) {
+                throw new Error("should never come here"); // TOTP can't work without MFA
+            }
+
+            const sessionUser = await getUser(session.getUserId(), userContext);
+            if (sessionUser === undefined) {
+                throw new SessionError({
+                    type: SessionError.UNAUTHORISED,
+                    message: "Session user not found",
+                });
+            }
+
+            await MultiFactorAuth.assertAllowedToSetupFactorElseThrowInvalidClaimError(session, "totp", userContext);
+
+            return await options.recipeImplementation.createDevice({
+                userId,
+                userIdentifierInfo,
+                deviceName: deviceName,
+                userContext: userContext,
+            });
+        },
+
+        listDevicesGET: async function ({ options, session, userContext }) {
+            const userId = session.getUserId();
+
+            return await options.recipeImplementation.listDevices({
+                userId,
+                userContext,
+            });
+        },
+
+        removeDevicePOST: async function ({ deviceName, options, session, userContext }) {
+            const userId = session.getUserId();
+
+            return await options.recipeImplementation.removeDevice({
+                userId,
+                deviceName,
+                userContext,
+            });
+        },
+
+        verifyDevicePOST: async function ({ deviceName, totp, options, session, userContext }) {
+            const userId = session.getUserId();
+            const tenantId = session.getTenantId();
+
+            const mfaInstance = MultiFactorAuthRecipe.getInstance();
+
+            if (mfaInstance === undefined) {
+                throw new Error("should never come here"); // TOTP can't work without MFA
```

Comment:
```suggestion
                throw new Error("Please initialise the MFA recipe as well."); // TOTP can't work without MFA
```

Reply:
It should never come here because we auto-init MFA. Updated the comment instead of updating the message.